lib/buffer.js

```diff
@@ -379,6 +378,16 @@ Buffer.prototype.toString = function(encoding, start, end) {
 };
 
 
+Buffer.prototype.toString = function() {
+  const length = this.length | 0;
+  if (length === 0)
+    return '';
+  if (arguments.length === 0)
+    return this.utf8Slice(0, length);
+  return slowToString.apply(this, arguments);
```

Comment:
@mraleph gave a talk explaining that V8 optimizes when `arguments` is passed to `.apply()`. It won't materialize the `arguments` object in this case, and is the fastest of the alternatives.